cmd/neofs-cli/modules/object/util_session_v2.go

```diff
@@ -0,0 +1,120 @@
+package object
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+
+	"github.com/nspcc-dev/neofs-node/cmd/neofs-cli/internal/common"
+	"github.com/nspcc-dev/neofs-node/cmd/neofs-cli/internal/commonflags"
+	"github.com/nspcc-dev/neofs-sdk-go/client"
+	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
+	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
+	"github.com/nspcc-dev/neofs-sdk-go/session/v2"
+	"github.com/nspcc-dev/neofs-sdk-go/user"
+	"github.com/spf13/cobra"
+)
+
+func getSessionV2(cmd *cobra.Command) (*session.Token, error) {
+	common.PrintVerbose(cmd, "Trying to read V2 session from the file...")
+
+	path, _ := cmd.Flags().GetString(commonflags.SessionToken)
+	if path == "" {
+		common.PrintVerbose(cmd, "File with session token is not provided.")
+		return nil, nil
+	}
+
+	common.PrintVerbose(cmd, "Reading V2 session from the file [%s]...", path)
+
+	var tok session.Token
+
+	err := common.ReadBinaryOrJSON(cmd, &tok, path)
+	if err != nil {
+		return nil, fmt.Errorf("read V2 session: %w", err)
+	}
+
+	return &tok, nil
+}
+
+func getVerifiedSessionV2(cmd *cobra.Command, cmdVerb session.Verb, key *ecdsa.PrivateKey, cnr cid.ID) (*session.Token, error) {
+	tok, err := getSessionV2(cmd)
+	if err != nil || tok == nil {
+		return tok, err
+	}
+
+	common.PrintVerbose(cmd, "Validating V2 session token...")
+
+	if err := tok.Validate(); err != nil {
+		return nil, fmt.Errorf("invalid V2 session token: %w", err)
+	}
+
+	if !tok.AssertVerb(cmdVerb, cnr) {
+		return nil, fmt.Errorf("v2 session token does not authorize %v for container %s", cmdVerb, cnr.String())
+	}
+
+	signer := user.NewAutoIDSigner(*key)
+	if tok.Issuer() != signer.UserID() {
+		return nil, fmt.Errorf("v2 session token issuer %v does not match user %s", tok.Issuer(), signer.UserID().String())
```

Comment:
`... does not match provided key/wallet`? i think a user with no understanding of source code will not understand what the signer is, and why it does not match. he has provided the key/wallet